### DISCUSSION TITLE
로깅을 어떻게 해야할까요?

### DISCUSSION THREAD

**Discussion by awesomeo184** [woowacourse-teams/2022-gong-check#391] [design]
무엇을 로깅해야할지, 로깅 시스템을 어떻게 구축해야할지 논의해봅시다.

**Comment by awesomeo184** [comment]
로깅해야할 것, 로깅하지 말아야 할 것에 대한 자료

https://www.freecodecamp.org/news/how-to-use-logs-effectively-in-your-code/

**Comment by awesomeo184** [comment]
현재 로깅 기준
Info - 사용자가 E2E 에서 겪을 수 있는 예외 상황 + 강조하고 싶은 로직
Warn - Postman 또는 curl 과 같은 현재 E2E 구조 상 이루어질 수 없는 예외
Error - 외부 API 예외

현재 잡아놓은 기준은 위와 같습니다.
하지만, 로깅을 적용하면서 마주친 문제점들이 존재합니다.
예를 들어, Warn 같은 경우, 프론트에서 잡아줄 수 없는 예외에 대해 Warn 로그를 찍도록 기준을 잡은 상태입니다.
여기서 생각해 볼 수 있는 것이, 어? "이거 프론트에서 예외 처리 하나?" 와 같은 프론트 의존적인 로깅이 되는 것 같아, 이에 대해 논의해보고자 합니다.

만약, 서버 입장에서만 생각한 로깅이라면, 현재 Info 와 Warn 으로 구분 지은 기준을 Info 로 합치고, Error 를 Warn 으로 구성하는 것이 어떤지 궁금합니다.

**Comment by jinyoungchoi95** [comment]
레벨 | 설명
-- | --
Fatal | 넌 이미 죽어있다
Error | 의도하지 않은 에러가 발생한 경우 (ex. Github 에 엑세스 토큰을 받는 과정에서 credential 이 잘못된 경우) 이건 우리가
Warn | 명확한 의도가 있는 에러 (ex. NotFoundException, Apache에서 제공한 로깅 그거를 봅시다.) 시스템 구조 상 클라이언트가 잘못 보낼 수 가 없는데 누군가 POSTMAN 으로 헛짓거리를 한다. 이건 나쁜거다.
Info | 요구사항에 따라 시스템 동작을 보여줄 때 대략적인 수준에서 응용 프로그램의 진행상황을 강조하는 정보 메시지로 사용
Debug | Info 레벨보다 더 자세한 정보가 필요한 경우. Dev 환경
Trace | Debug 레벨보다 더 자세함. Dev 환경에서 버그를 해결하기 위해 사용

**Comment by meatsby** [comment]
`Info` - 거의 사용 X
`Warn` - 모든 사용자 예외
`Error` - Infrastructure 예외 및 예상하지 못한 예외
